support-workers/src/main/scala/com/gu/salesforce/SalesforceService.scala

```diff
@@ -51,15 +51,24 @@ class SalesforceService(config: SalesforceConfig, client: FutureHttpClient)(impl
       recipientContactRecord <- maybeAddGiftRecipient(
         contactRecord.ContactRecord,
         giftRecipient,
-        user.deliveryAddress.getOrElse(user.billingAddress)
+        user
       )
     } yield SalesforceContactRecordsResponse(contactRecord, recipientContactRecord)
   }
 
-  private def maybeAddGiftRecipient(contactRecord: SalesforceContactRecord, maybeGiftRecipient: Option[GiftRecipient], deliveryAddress: Address) =
+  private def deliveryAddressForGiftRecipientType(giftRecipient: GiftRecipient, user: User) =
+    giftRecipient match {
+      case _: GiftRecipient.WeeklyGiftRecipient => Some(user.deliveryAddress.getOrElse(user.billingAddress))
```

Comment:
Should we just use deliveryAddress here rather than falling back the billingAddress if it's None?

Reply:
agree we should, I suspect that it is "not possible" but someone added that to get rid of the Optionality in the types rather than doing a `.get` or throwing an exception.
When we turn all the State case classes into traits like we did for the email state, the delivery address would not be optional for GW and wouldn't exist for DS so this problem will not exist.

Reply:
Agreed

Reply:
Yeah if there's a bug caused by the [] Use billing address as delivery address feature (or vice-versa) then it should be reconciled at a higher level not just left to be rescued via this random fallback.